src/components/LinkIncident/Link.tsx

```diff
@@ -0,0 +1,155 @@
+import {
+  AnyIcon,
+  Button,
+  Stack,
+  useInitialisedDeskproAppClient,
+} from "@deskpro/app-sdk";
+import { useQueryWithClient } from "../../hooks/useQueryWithClient";
+import { getIncidents } from "../../api/api";
+import { useState } from "react";
+import useDebounce from "../../hooks/debounce";
+import { Checkbox, Input } from "@deskpro/deskpro-ui";
+import { faMagnifyingGlass } from "@fortawesome/free-solid-svg-icons";
+import { HorizontalDivider } from "../HorizontalDivider/HorizontalDivider";
+import IncidentJson from "../../mapping/incident.json";
+import { FieldMapping } from "../FieldMapping/FieldMapping";
+import { useLinkIncidents, useTicketCount } from "../../hooks/hooks";
+import { LoadingSpinnerCenter } from "../LoadingSpinnerCenter/LoadingSpinnerCenter";
+
+export const LinkIncident = () => {
+  const [inputText, setInputText] = useState<string>("");
+  const [selectedIncidents, setSelectedIncidents] = useState<string[]>([]);
+  const [linkedIncidents, setLinkedIncidents] = useState<string[]>([]);
+  const [incidentLinketCount, setIncidentLinkedCount] = useState<
+    Record<string, number>
+  >({});
+  const [page, setPage] = useState<number>(0);
+
+  const { debouncedValue: debouncedText } = useDebounce(inputText, 300);
+  const { getLinkedIncidents, linkIncidents } = useLinkIncidents();
+  const { getMultipleIncidentTicketCount } = useTicketCount();
+
+  useInitialisedDeskproAppClient((client) => {
+    client.setTitle("Link Incident");
+  }, []);
+
+  const incidentsQuery = useQueryWithClient(
+    ["getIncidents", page, debouncedText],
+    (client) => getIncidents(client, page, debouncedText),
+    {
+      onSuccess: async (data) => {
+        const linkedIncidentsFunc = await getLinkedIncidents();
+
+        if (!linkedIncidentsFunc) return;
+
+        const linkedIncidentsIds = data?.incidents
+          ?.filter((item) => linkedIncidentsFunc.includes(item.id))
+          .map((e) => e.id);
+
+        const linkedIncidentTickets = await getMultipleIncidentTicketCount(
+          linkedIncidentsIds
+        );
+
+        setLinkedIncidents([...linkedIncidents, ...linkedIncidentsIds]);
+
+        setIncidentLinkedCount({
+          ...incidentLinketCount,
+          ...linkedIncidentTickets,
+        });
+      },
+    }
+  );
+
+  const incidentsData = incidentsQuery.data;
+
+  return (
+    <Stack gap={10} style={{ width: "100%" }} vertical>
+      <Input
+        onChange={(e) => setInputText(e.target.value)}
+        value={inputText}
+        placeholder="Enter Incident Key"
+        type="text"
+        leftIcon={faMagnifyingGlass as AnyIcon}
+      />
+      {!incidentsQuery.isSuccess || !incidentLinketCount ? (
+        <LoadingSpinnerCenter />
```

Comment:
if nothing is found, the loader spinning. How user understand that nothing was found or that the download is still in progress?